tests/Opus/Search/Common/IndexerTest.php

```diff
@@ -0,0 +1,961 @@
+ * details. You should have received a copy of the GNU General Public License
+ * along with OPUS; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * @category    Test
+ * @package     Opus_Search
+ * @author      Thomas Urban <thomas.urban@cepharum.de>
+ * @copyright   Copyright (c) 2010-2015, OPUS 4 development team
+ * @license     http://www.gnu.org/licenses/gpl.html General Public License
+ * @version     $Id$
+ */
+
+/**
+ * Test indexing.
+ *
+ */
+class Opus_Search_Common_IndexerTest extends TestCase {
+
+	/**
+	 * @var Opus_Search_Indexing
+	 */
+	protected $indexer;
+
+	/**
+	 * @var int
+	 */
+	protected $document_id;
+
+	/**
+	 * @var string
+	 */
+	protected $files_dir;
+
+	/**
+	 * @var Zend_Config
+	 */
+	protected $config;
+
+	/**
+	 * @var Opus_Document
+	 */
+	protected $nullDoc;
+
+	/**
+	 * Valid document data.
+	 *
+	 * @var array  An array of arrays of arrays. Each 'inner' array must be an
+	 * associative array that represents valid document data.
+	 */
+	protected static $_validDocumentData = array(
+		'Type'                    => 'article',
+		'Language'                => 'deu',
+		'ContributingCorporation' => 'Contributing, Inc.',
+		'CreatingCorporation'     => 'Creating, Inc.',
+		'ThesisDateAccepted'      => '1901-01-01',
+		'Edition'                 => 2,
+		'Issue'                   => 3,
+		'Volume'                  => 1,
+		'PageFirst'               => 1,
+		'PageLast'                => 297,
+		'PageNumber'              => 297,
+		'CompletedYear'           => 1960,
+		'CompletedDate'           => '1901-01-01',
+	);
+
+	/**
+	 * Valid document data provider
+	 *
+	 * @return array
+	 */
+	public static function validDocumentDataProvider() {
+		return self::$_validDocumentData;
+	}
+
+	/**
+	 * This method is called before a test is executed.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		$this->files_dir = Opus_Config::get()->workspacePath . '/files';
+
+		$this->indexer = Opus_Search_Service::selectIndexingService();
+
+		$document = new Opus_Document();
+		foreach ( self::$_validDocumentData as $fieldname => $value ) {
+			$callname = 'set' . $fieldname;
+			$document->$callname( $value );
+		}
+		$document->store();
+		$this->document_id = $document->getId();
+	}
+
+	/**
+	 * This method is called after a test is executed.
+	 */
+	protected function tearDown() {
+		// remove test documents under tests/workspace/files/$document_id
+		$dirname = $this->files_dir . DIRECTORY_SEPARATOR . $this->document_id;
+		if ( is_dir( $dirname ) && is_readable( $dirname ) ) {
+			foreach ( glob( $dirname . "/*" ) as $filename ) {
+				if ( is_readable( $filename ) ) {
+					unlink( $filename );
+				}
+			}
+			rmdir( $dirname );
+		}
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @expectedException Opus_Search_InvalidConfigurationException
+	 */
+	public function testMissingConfigParamSearchEngine_Index_Host() {
```

Comment:
Are there other tests that check what happens if configuration is missing? I am a bit worried about changed behavior that I don't even know about.